hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/SCMRatisServerImpl.java

```diff
@@ -429,4 +430,14 @@ public RaftPeer getLeader() {
           division.getRaftConf().getPeer(RaftPeerId.valueOf(leaderId));
     }
   }
+
+  @Override
+  @SuppressWarnings("unchecked")
+  public <T> T getProxyHandler(final RequestType type, final Class<T> intf, final T impl) {
+    final SCMHAInvocationHandler invocationHandler =
+        new SCMHAInvocationHandler(type, impl, this);
+    return (T) Proxy.newProxyInstance(getClass().getClassLoader(),
+        new Class<?>[] {intf}, invocationHandler);
+  }
```

Comment:
- This can be moved to `SCMRatisServer` as `default` method, then we don't need to duplicate in `SCMHAManagerStub`.
- Suppressed unchecked warning can be fixed by replacing `(T)` with: `intf.cast(...)`.
- Should we keep using `SCMHAInvocationHandler.class.getClassLoader()`?  (In practice I don't think these would give different results.)

Reply:
Thanks @adoroszlai for the detailed review.
I somehow missed `FinalizationStateManagerImpl`, `SCMCertStore` & `ScmSecretKeyStateBuilder`, thanks for catching them.

Addressed all the review comments.